File: app/src/main/java/rebus/navutils/sample/MainActivity.java
package rebus.navutils.sample;

import android.os.Bundle;
import android.view.View;

import rebus.navutils.NavUtils;
import rebus.navutils.activity.ActivityUtils;
import rebus.navutils.activity.BaseActivity;
import rebus.navutils.chrome.ChromeUtils;

public class MainActivity extends BaseActivity implements View.OnClickListener {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setTitle("NavUtils Utils");
        setSubtitle("MainActivity");
        setIcon(R.mipmap.ic_launcher);
        setFinishOnNavigationOnClickListener();
        setStatusBarColorRes(R.color.colorAccent);
        setLightStatusBar();
        findViewById(R.id.horizontal_right).setOnClickListener(this);
        findViewById(R.id.horizontal_left).setOnClickListener(this);
        findViewById(R.id.vertical_bottom).setOnClickListener(this);
        findViewById(R.id.vertical_top).setOnClickListener(this);
        findViewById(R.id.none).setOnClickListener(this);
        findViewById(R.id.system).setOnClickListener(this);
        findViewById(R.id.chrome).setOnClickListener(this);
    }

    @Override
    protected int setLayoutResource() {
        return R.layout.activity_main;
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.horizontal_right:
                ActivityUtils.Builder(getActivity())
                        .animationType(NavUtils.HORIZONTAL_RIGHT)
                        .start(ResultActivity.class);
                break;
            case R.id.horizontal_left:
                ActivityUtils.Builder(getActivity())
                        .animationType(NavUtils.HORIZONTAL_LEFT)
                        .start(ResultActivity.class);
                break;
            case R.id.vertical_bottom:
                ActivityUtils.Builder(getActivity())
                        .animationType(NavUtils.VERTICAL_BOTTOM)
                        .start(ResultActivity.class);
                break;
            case R.id.vertical_top:
                ActivityUtils.Builder(getActivity())
                        .animationType(NavUtils.VERTICAL_TOP)
                        .start(ResultActivity.class);
                break;
            case R.id.none:
                ActivityUtils.Builder(getActivity())
                        .animationType(NavUtils.NONE)
                        .start(ResultActivity.class);
                break;
            case R.id.system:
                ActivityUtils.Builder(getActivity())
                        .animationType(NavUtils.SYSTEM)
                        .start(ResultActivity.class);
                break;
            case R.id.chrome:
                ChromeUtils.Builder(getActivity())
                        .setShowTitle(true)
                        .animationType(NavUtils.VERTICAL_BOTTOM)
                        .setToolbarColorRes(R.color.colorAccent)
                        .load("https://github.com/rebus007/NavUtils");
                break;
        }

    }
}

File: sample/src/main/java/rebus/navutils/sample/MainActivity.kt
package rebus.navutils.sample

import android.content.pm.ActivityInfo
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*
import rebus.navutils.NavUtils
import rebus.navutils.activity.BaseActivity
import rebus.navutils.pushActivity
import rebus.navutils.pushChromeCustomTab

class MainActivity : BaseActivity(), View.OnClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setTitle("NavUtils Utils")
        setSubtitle("MainActivity")
        setIcon(R.mipmap.ic_launcher)
        setFinishOnNavigationOnClickListener()
        setStatusBarColorRes(R.color.colorAccent)
        setLightStatusBar()
        horizontalRight.setOnClickListener(this)
        horizontalLeft.setOnClickListener(this)
        verticalBottom.setOnClickListener(this)
        verticalTop.setOnClickListener(this)
        none.setOnClickListener(this)
        system.setOnClickListener(this)
        chrome.setOnClickListener(this)
        sceneTransition.setOnClickListener(this)
    }

    override fun setLayoutResource() = R.layout.activity_main

    override fun setRequestedOrientation() = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT

    override fun onClick(v: View) {
        when (v.id) {
            R.id.horizontalRight -> {
                pushActivity(ResultActivity::class) {
                    animationType(NavUtils.Anim.HORIZONTAL_RIGHT)
                }.commit()
            }
            R.id.horizontalLeft -> {
                pushActivity(ResultActivity::class) {
                    animationType(NavUtils.Anim.HORIZONTAL_LEFT)
                }.commit()
            }
            R.id.verticalBottom -> {
                pushActivity(ResultActivity::class) {
                    animationType(NavUtils.Anim.VERTICAL_BOTTOM)
                }.commit()
            }
            R.id.verticalTop -> {
                pushActivity(ResultActivity::class) {
                    animationType(NavUtils.Anim.VERTICAL_TOP)
                }.commit()
            }
            R.id.none -> {
                pushActivity(ResultActivity::class) {
                    animationType(NavUtils.Anim.NONE)
                }.commit()
            }
            R.id.system -> {
                pushActivity(ResultActivity::class) {
                    animationType(NavUtils.Anim.SYSTEM)
                }.commit()
            }
            R.id.chrome -> {
                pushChromeCustomTab {
                    animationType(NavUtils.Anim.VERTICAL_BOTTOM)
                }.load("https://github.com/raphaelbussa/NavUtils")
            }
            R.id.sceneTransition -> {
                pushActivity(SceneTransitionActivity::class) {
                    animationType(NavUtils.Anim.SYSTEM)
                }.commit()
            }
        }
    }

}
